bin/varnishd/mgt/mgt_jail_linux.c

```diff
@@ -48,11 +48,65 @@
 #include "mgt/mgt.h"
 #include "common/heritage.h"
 
+static int
+vjl_set_thp(const char *arg)
+{
+	int val;
+
+	if (!strcmp(arg, "ignore"))
+		return (0);
+	if (!strcmp(arg, "enable"))
+		val = 0;
+	else if (!strcmp(arg, "disable"))
+		val = 1;
+	else {
+		ARGV_ERR(
+		    "linux jail: unknown value '%s' for argument transparent_hugepage.\n",
+		    arg);
+	}
+	if (prctl(PR_SET_THP_DISABLE, val, 0, 0, 0) != 0) {
+		MGT_Complain(C_ERR,
+		    "Could not %s Transparent Hugepage: %s (%d)",
+		    arg, VAS_errtxt(errno), errno);
+	} else {
+		MGT_Complain(C_INFO, "linux jail: Transparent Hugepage now %s",
+		    prctl(PR_GET_THP_DISABLE, 0, 0, 0, 0) ?
+		    "disabled" : "enabled");
```

Comment:
Not sure we need to say anything upon success. If we really want to, then this format is probably enough:

```c
"linux jail: Transparent Hugepage %sd", arg
```

We tend to not log much when something succeeds (there are cases where we do) so I'm interested in @nigoroll's opinion on this one.

Reply:
This was done following @asadsa92 's comment https://github.com/varnishcache/varnish-cache/pull/4174#discussion_r1767002337 I guess we'll need his opinion too then :)

Reply:
I think there is a misunderstanding, I agree we do not need to print anything for the success case.
But, `PR_GET_THP_DISABLE` was meant to retrieve whatever the current value of THP is from the kernel.
If we set it, then I agree we do not need to print the value. The kernel already reported a success there.
However, the "ignore" case should retrieve and print the current value.

Reply:
If you print the value when the configuration is `ignore` but fail to retrieve the value then you also have to prepare an error message for that.

I'm not seeing this being worth the trouble and I'd rather actually ignore THP entirely, and not care about its current state. We leave it as an exercise for the operator, but we let them know when they picked a state and it couldn't be applied.